report/6.Conclusion/Perspectivation.tex

```diff
@@ -1 +1,103 @@
+
+\subsection{Test cases}
+Along with the MoSCoW analysis, four test cases were presented: 
+\begin{itemize}
+    \item Dropping a balloon
+    \item Rolling a ball
+    \item Throwing a ball
+    \item Shooting a balloon
+\end{itemize}
+
+The first case, dropping a ball, was already presented in the MoSCoW conclusion and deemed successful.
+The test was conducted with a heart shaped, red balloon with a diameter of 30 centimeters, which was dropped at a distance of 4 meters and a height of 2 meters. 
+\texttt{F.L.A.T.} was able to keep it's laser on the target most of the time during the fall. 
+For the rolling a ball test case, \texttt{F.L.A.T.} was able to roughly keep it's laser on a ball with a diameter of 52 mm successfully. 
+However, the final two cases with a throwing a ball and shooting a balloon were not accomplished, as the device was not fast enough to process the speed of the object, and does not have the capabilities to shoot a balloon.
+
+\subsection{Machine intelligence}
+Several observations were made during the development of the MI aspect of the solution.
+Initially, the idea for the project was to recognize objects in real-time using machine learning, however as described in \autoref{des:sec:performance}, the object recognition algorithms available, were not performant enough given the hardware limitations.
+Instead, the variable motor power problem, described in \autoref{subsec:variable_motor_power}, turned out to be an excellent way to utilize MI for the solution.
+The calibration module was implemented using a neural network, which was executed on the NXT.
+This proved to be an excellent solution to the variable motor power problem, and a great way to use MI in the solution.
+
+An important lesson learned from this project was that the usage of MI required thought about the domain in which the MI is applied to.
+
+
+\subsection{Real-time requirements}
+For the real-time part of the system, several observations were made during the development of the project.
+First of all, the amount of tasks that were running on the NXT in order to make it functional were far from enough to give a good CPU utilization using the cyclic executive scheduling model.
+In order to optimize the utilization time, parts of the machine intelligence calculations responsible for calculating the amount of power required to move were moved to the NXT for execution.
+This resulted in the cyclic executive model being unfeasible, as there was a task that took more than 1 millisecond to run while another task had a deadline of 1 millisecond, thus making it unschedulable. 
+To fix this, the scheduling model was changed to fixed priority scheduling with preemption.
+The optimal choice for a system like this would be the earliest deadline first scheduling model, but this is not implementable in the NXTOSEK operating system, as this is a fixed priority operating system.
```

Comment:
```suggestion
The optimal choice for a system like this would be the earliest deadline first scheduling model, but this is not implementable in the nxtOSEK operating system, as this is a fixed priority operating system.
```